developer-hub/src/reference_docs/extrinsic_encoding.rs

```diff
@@ -0,0 +1,156 @@
+//! variant.
+//!
+//! ### signature
+//!
+//! This is the SCALE encoded signature. The signature is obtained by signing the _signed payload_
+//! bytes (see below on how this is constructed) using the private key associated with the address,
+//! and a signing algorithm that can vary across different chains.
+//!
+//! The signature type used on the Polkadot relay chain is [`sp_runtime::MultiSignature`]; the
+//! variants there are the types of signature that can be provided.
+//!
+//! ### signed_extensions_extra
+//!
+//! This is the concatenation of the SCALE encoded bytes representing each of the
+//! [_signed extensions_][sp_runtime::traits::SignedExtension] that the chain is configured with,
+//! in the order that they are configured for that chain. Signed extensions are, briefly, a means
+//! for different chains to extend the "basic" extrinsic format with custom data that can be checked
+//! by the runtime.
+//!
+//! When it comes to constructing an extrinsic, each signed extension has two things that we are
+//! interested in here:
+//!
+//! - The actual SCALE encoding of the signed extension type itself; this is what will form our
+//!   `signed_extensions_extra` bytes.
+//! - An `AdditionalSigned` type. This is SCALE encoded into the `signed_extensions_additional`
+//!   data of the _signed payload_ (see below).
+//!
+//! Either (or both) of these can encode to zero bytes.
+//!
+//! Each chain configures the set of signed extensions that it uses in its runtime configuration.
+//! At the time of writing, Polkadot configures them
+//! [here](https://github.com/polkadot-fellows/runtimes/blob/1dc04eb954eadf8aadb5d83990b89662dbb5a074/relay/polkadot/src/lib.rs#L1432C25-L1432C25).
+//! Some of the common signed extensions are defined [here][frame::deps::frame_system].
+//!
+//! Information about exactly which signed extensions are present on a chain and in what order is
+//! also a part of the metadata for the chain. For V15 metadata, it can be
+//! [found here][frame::deps::frame_support::__private::metadata::v15::ExtrinsicMetadata].
+//!
+//! ## call_data
+//!
+//! This data defines exactly which call is made by the extrinsic, and with what arguments, ie:
+//!
+//! ```text
+//! call_data = concat(
+//!     pallet_index,
+//!     call_index,
+//!     call_args
+//! )
+//! ```
+//!
+//! - `pallet_index` is a single byte denoting the index of the pallet that we are calling into.
+//! - `call_index` is a single byte denoting the index of the call that we are making the pallet.
+//! - `call_args` are the SCALE encoded bytes for each of the arguments that the call expects.
+//!
+//! Information about the pallets that exist for a chain (including their indexes), the calls
+//! available in each pallet (including their indexes), and the arguments required for each call
+//! can be found in the metadata for the chain. For V15 metadata, this information
+//! [is here][frame::deps::frame_support::__private::metadata::v15::PalletMetadata].
```

Comment:
In acouple of places I wanted to point to `frame-metadata`. Better to add as a direct dependency of this developer-hub crate or perhaps expose more easily via frame-support?

Reply:
I would leave it as-is for now to not distract us, but I expect the `frame` crate to properly expose all relevant types long term to avoid needing to go down these hidden paths.